docs/ARCHITECTURE.md

```diff
@@ -217,19 +306,34 @@ classDiagram
 direction LR
     namespace Ai {
         class AiClient {
-            +prompt(?string $text) PromptBuilder$
-            +message(?string $text) MessageBuilder$
+            +prompt(...string $text) PromptBuilder$
+            +message(...string $text) MessageBuilder$
         }
 
         class PromptBuilder {
             +withText(string $text) self
+            +withImage(string $mimeType, string $base64Blob) self
             +withImageFile(File $file) self
             +withAudioFile(File $file) self
             +withVideoFile(File $file) self
             +withFunctionResponse(FunctionResponse $functionResponse) self
             +withMessageParts(...MessagePart $part) self
             +withHistory(...Message $messages) self
             +usingModel(AiModel $model) self
+            +usingModelSupporting(...AiCapability|AiOption $aiCapabilityOrOption) self
+            +usingModelSupportingCapability(...AiCapability $aiCapability) self
+            +usingModelSupportingOption(...AiOption $aiOption) self
+            +usingModelSupportingAudio() self
+            +usingModelSupportingHistory() self
+            +usingModelSupportingEmbeddings() self
+            +usingModelSupportingImages() self
+            +usingModelSupportingJsonOutput() self
+            +usingModelSupportingMusic() self
+            +usingModelSupportingOutputSchema() self
+            +usingModelSupportingSpeech() self
+            +usingModelSupportingText() self
+            +usingModelSupportingTextToSpeech() self
+            +usingModelSupportingVideo() self
```

Comment:
See my comments above, I'm not sure these methods are needed since the required model capabilities and options are always inferred from the prompt. There is already a way to verbosely handle this "manually", IMO no need to overcomplicate the fluent API for something that's unnecessary.